plugins/chat/assets/javascripts/discourse/components/toggle-channel-membership-button.js

```diff
@@ -1,53 +1,46 @@
-import Component from "@ember/component";
+import Component from "@glimmer/component";
 import I18n from "I18n";
 import { inject as service } from "@ember/service";
 import { popupAjaxError } from "discourse/lib/ajax-error";
-import { action, computed } from "@ember/object";
-
+import { action } from "@ember/object";
+import { tracked } from "@glimmer/tracking";
 export default class ToggleChannelMembershipButton extends Component {
   @service chat;
-
-  tagName = "";
-  channel = null;
+  @tracked isLoading = false;
   onToggle = null;
-  options = null;
-  isLoading = false;
+  options = {};
 
-  init() {
-    super.init(...arguments);
+  constructor() {
+    super(...arguments);
 
-    this.set(
-      "options",
-      Object.assign(
-        {
-          labelType: "normal",
-          joinTitle: I18n.t("chat.channel_settings.join_channel"),
-          joinIcon: "",
-          joinClass: "",
-          leaveTitle: I18n.t("chat.channel_settings.leave_channel"),
-          leaveIcon: "",
-          leaveClass: "",
-        },
-        this.options || {}
-      )
+    this.options = Object.assign(
+      {
+        labelType: "normal",
+        joinTitle: I18n.t("chat.channel_settings.join_channel"),
+        joinIcon: "",
+        joinClass: "",
+        leaveTitle: I18n.t("chat.channel_settings.leave_channel"),
+        leaveIcon: "",
+        leaveClass: "",
+      },
+      this.args.options || {}
     );
```

Comment:
```suggestion
    this.options = {
      labelType: "normal",
      joinTitle: I18n.t("chat.channel_settings.join_channel"),
      joinIcon: "",
      joinClass: "",
      leaveTitle: I18n.t("chat.channel_settings.leave_channel"),
      leaveIcon: "",
      leaveClass: "",
      ...this.args.options
    },
```
(this also works fine when `args.options` is `null`/`undefined`)